### ISSUE TITLE
When the query time span of logs is relatively large (such as 30 days), and the number of logs is relatively large, it cannot be queried

### ISSUE THREAD

**Issue by xingdingchun** [grafana/loki#4377]
Error message
level=warn ts=2021-09-24T08:14:15.613284568Z caller=logging.go:71 traceID=46c94d4df121a451 msg="GET /loki/api/v1/query_range?query=%7Bapp%3D%22prod-play%22%7D+%7C%3D%22686642%22+%7C%3D%22base-info%22&start=1629820800.0 (504) 5m0.009197849s Response: \"Request timed out, decrease the duration of the request or add more label matchers (prefer exact match over regex match) to reduce the amount of data processed.\\n\" ws: false; Accept: */*; Accept-Encoding: gzip, deflate; Connection: close; User-Agent: python-requests/2.24.0; X-Forwarded-For: 103.82.4.109; X-Real-Ip: 103.82.4.109; "

How to solve this problem? Or you can configure parameters to solve this problem？

**Comment by xingdingchun** [comment]
If you use grafana to access, the error message is as follows:
level=info ts=2021-09-24T08:03:44.427432179Z caller=metrics.go:91 org_id=fake traceID=110a4c9bf7e6dad8 latency=slow query="{app=\"prod-play\"} |= \"93.70.103.211\" |= \"by_password\"" query_type=filter range_type=range length=24h0m1s step=30s duration=38.322324594s status=200 limit=1000 returned_lines=0 throughput=268MB total_bytes=10GB
level=info ts=2021-09-24T08:04:14.289255592Z caller=metrics.go:91 org_id=fake traceID=160ead3b293992c1 latency=slow query="{app=\"prod-play\"} |= \"93.70.103.211\" |= \"by_password\"" query_type=filter range_type=range length=24h0m1s step=30s duration=38.058951342s status=200 limit=1000 returned_lines=0 throughput=270MB total_bytes=10GB
level=info ts=2021-09-24T08:04:43.68347307Z caller=metrics.go:91 org_id=fake traceID=1a7744101f2d4e9c latency=slow query="{app=\"prod-play\"} |= \"93.70.103.211\" |= \"by_password\"" query_type=filter range_type=range length=24h0m1s step=30s duration=37.331194751s status=200 limit=1000 returned_lines=0 throughput=276MB total_bytes=10GB
level=info ts=2021-09-24T08:05:30.071108862Z caller=table_manager.go:324 msg="synching tables" expected_tables=1

**Comment by benridley** [comment]
@cyriltovena I'm running a standalone instance of Loki and have noticed it's only using a single thread for my queries which is performing pretty badly. Is there a way to achieve some level of parallelism at the process level, or can this only be achieved by splitting the query over intervals using multiple query frontends?

**Comment by benridley** [comment]
I found a thread that says Loki 2.4 will include parallel queries on a single instance, so I guess this might be solved!

**Comment by patsevanton** [comment]
@benridley How are you fix this issue ?

**Comment by benridley** [comment]
> @benridley How are you fix this issue ?

The main thing that helped me was increasing RAM significantly and also enabling swap on my Loki server (I was using a single instance with no swap space). Once I did this, things worked smoothly. It’s also worth playing with the settings for the intervals used in querying to make sure you aren’t double handling when downloading chunks from your object store. I also switched the compression algorithm for chunks to lz4 which is easier to decompress.
